src/pages/logs-page.js

```diff
@@ -99,16 +99,17 @@ export function LogsPage() {
     };
 
     getLogs(queryBody).then((res) => {
-      if (res.data != null) {
-        setLogs(res.data.results);
-      }
+      setLogs(res?.data?.results || []);
       setLogsIsLoading(false);
     });
   }, []);
 
   useEffect(() => {
-    loadLogs();
-  }, [loadLogs, topologyId]);
+    const subscription = watch(() => {
+      loadLogs();
+    });
+    return () => subscription.unsubscribe();
+  }, [watch]);
```

Comment:
We don't need to watch for `topologyId` here? Including `loadLogs` in should also be ok?

Reply:
we are watching all controls changes here so topologyId is also covered. loadlogs not required as it's reference won't change.